libsolidity/formal/CHC.cpp

```diff
@@ -511,6 +511,31 @@ void CHC::endVisit(IndexRangeAccess const& _range)
 	m_context.addAssertion(sliceArray->length() == end - start);
 }
 
+void CHC::endVisit(Return const& _return)
+{
+	SMTEncoder::endVisit(_return);
+
+	connectBlocks(m_currentBlock, predicate(*m_returnDests.back()));
+
+	auto returnGhost = createBlock(&_return, PredicateType::FunctionBlock, "return_ghost_");
+	m_currentBlock = predicate(*returnGhost);
```

Comment:
What is the purpose of these ghost blocks?
I am checking a simple program and there it seems like there are rules with `return_ghost` in the body, but no rules with `return_ghost ` in the head, so in that example they seem useless.
When are they important?

Reply:
They are used for dead code after `return` which might happen. Example:
```
if (cond) {
    x = 2;
    return;
    x = 3;
}
```
When we visit `return`, we add the edge `m_currentBlock => return_dest_14` (or something like that), but we don't make `return_dest_14` the current block because we're not there yet. If we don't add the ghost block, `x = 3` would be added as a constraint when visited, after the return. This would then go in the next added rule from `m_currentBlock`.

Reply:
Briefly: the goal is precisely to create unreachable blocks containing the unreachable code post-return.

Reply:
> Briefly: the goal is precisely to create unreachable blocks containing the unreachable code post-return.

Right, I understand. Sounds good!
Would it be possible to add a comment explaining exactly that in the code?

Reply:
That would be nice. Than we can merge it.